core/src/main/java/io/jenkins/pluginhealth/scoring/probes/JSR305Probe.java

```diff
@@ -0,0 +1,43 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.util.List;
+
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * This probe checks for deprecated @Nonnull and @CheckForNull annotations in a plugin.
+ */
+@Component
+@Order(value = AbstractDeprecatedAnnotationImportsCheckProbe.ORDER)
+public class JSR305Probe extends AbstractDeprecatedAnnotationImportsCheckProbe {
+    public static final String KEY = "JSR-305";
+
+    @Override
+    public String key() {
+        return KEY;
+    }
+
+    @Override
+    public String getDescription() {
+        return "The probe checks for deprecated annotations.";
+    }
+
+    @Override
+    String getSuccessMessage() {
+        return "JSR305 annotation is updated";
+    }
+
+    @Override
+    String getFailureMessage() {
+        return "Deprecated annotations found";
+    }
+
+    @Override
+    public List<String> getListOfDeprecatedImports() {
+        return List.of(
+            "import javax.annotation.Nonnull;",
+            "import javax.annotation.CheckForNull;"
```

Comment:
If you're going for an implementation based on a `abstract` class that is supposed to deal with imports, you should not have to write the full `import` statement in here.

Just writing the fully qualified name of the import you are interested in, like `javax.annotation.Nonnull` should be enough, and it is up to the abstract class to then create the proper line with `import xxx;`.

(think that if you had to write another probe based on that, you would have to repeat writing `import xxx;` which is error prone and has no particular value).